### ISSUE TITLE
Bug Report: Network switching not working properly

### ISSUE THREAD

**Issue by skundu42** [Shardeum/shardeum-dapp-boilerplate#5]
Switching between Shardeum Liberty and Sphinx not working properly and even after confirming the metamask popup, the network is not switching.

Possible issue:
- Misconfiguration in setting up custom network setup in _app.tsx. Instead of using public provider, we need to use the [Jsonrpc provider](https://www.rainbowkit.com/docs/custom-chains).
- Also for easier debugging purpose console log the current network id.

**Comment by dhruvdabhi101** [comment]
I will fix this ASAP.

**Comment by dhruvdabhi101** [comment]
So, I added that jsonRpcProvider, and what happens is, when I change from Liberty to Sphinx... It changes normally, but when changing from Sphinx to Liberty, it stucks there.. It doesn't connect.

**Comment by tab21** [comment]
Hi 
Is anyone working on this issue ? 
If not can this issue be assigned to me please.

**Comment by skundu42** [comment]
Sure @tab21. I am assigning you this issue.

Minor updates: We have some changes. Wagmi now supports [Sphinx](https://wagmi.sh/react/chains) so you can use that directly instead of adding it as a custom chain.

**Comment by rahulsingh3526** [comment]
hey, I recently had an internship task to do at shardeum. I tried using the boilerplate but it wasn't working. So I made it from scratch. I did see and test out on other networks .like goerli and Polygon Mumbai . the switching is happening easily between these chains( and is being displayed on the frontend but when I change to liberty after going to goerli or Mumbai testnet .

 I noticed that inside metamask the network is changing. So if you just reload the page after going  from goerli to shardeum chains then it will display on the website. 

basically it changed inside the metamask but not on the display of the website. Could I become the assignee for this and try to resolve.
